lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -575,7 +583,13 @@ size_t _GenerateNetwork2TopoHash(const HdMaterialNetwork2& materialNetwork)
                 auto cmIt = node.parameters.find(cmName);
                 if (cmIt != node.parameters.end()) {
                     MayaUsd::hash_combine(topoHash, hash_value(cmIt->first));
-                    MayaUsd::hash_combine(topoHash, hash_value(cmIt->second));
+                    if (cmIt->second.IsHolding<TfToken>()) {
+                        auto const& colorSpace = cmIt->second.UncheckedGet<TfToken>();
+                        MayaUsd::hash_combine(topoHash, hash_value(colorSpace));
+                    } else if (cmIt->second.IsHolding<std::string>()) {
```

Comment:
That was the root of the problem we are trying to solve. The attribute can be a std::string or a TfToken, so the code needs to handle both.